scipy/spatial/transform/tests/test_rigid_transform.py

```diff
@@ -10,19 +10,26 @@
     xp_vector_norm,
     is_numpy,
     xp_assert_close,
+    make_xp_pytest_marks,
+    make_xp_test_case,
+    xp_assert_equal,
+    xp_promote
 )
 import scipy._lib.array_api_extra as xpx
 
+lazy_xp_modules = [RigidTransform, normalize_dual_quaternion]
 
-pytestmark = pytest.mark.skip_xp_backends(np_only=True)
+pytestmark = make_xp_pytest_marks(RigidTransform.as_matrix)
```

Comment:
I'm not a fan of this. This makes it quite hard to spot apparently unmarked tests, and has the unexpected side effect of marking non-array-API tests. I'd rather have a marker on each test although I would avoid unnecessary verbosity if there is already a much stricter marker. In other words, mark a test explicitly if it tests `as_matrix`; don't add a mark for `as_matrix` if the test tests something else (and there is already a `make_xp_pytest_marks` for it) and just uses as_matrix to fetch the output.